orca-pipelinetemplate/src/test/groovy/com/netflix/spinnaker/orca/pipelinetemplate/PipelineTemplatePipelinePreprocessorSpec.groovy

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipelinetemplate
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spectator.api.Clock
+import com.netflix.spectator.api.Registry
+import com.netflix.spectator.api.Timer
+import com.netflix.spinnaker.orca.extensionpoint.pipeline.PipelinePreprocessor
+import com.netflix.spinnaker.orca.pipelinetemplate.loader.FileTemplateSchemeLoader
+import com.netflix.spinnaker.orca.pipelinetemplate.loader.TemplateLoader
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.render.HandlebarsRenderer
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.render.Renderer
+import org.unitils.reflectionassert.ReflectionComparatorMode
+import spock.lang.Specification
+import spock.lang.Subject
+
+import static org.unitils.reflectionassert.ReflectionAssert.assertReflectionEquals
+
+class PipelineTemplatePipelinePreprocessorSpec extends Specification {
+
+  ObjectMapper objectMapper = new ObjectMapper()
+
+  TemplateLoader templateLoader = new TemplateLoader([new FileTemplateSchemeLoader(objectMapper)])
+
+  Renderer renderer = new HandlebarsRenderer(objectMapper)
+
+  Registry registry = Mock() {
+    clock() >> Mock(Clock) {
```

Comment:
There's a metric in the `PipelineTemplatePipelinePreprocessor` that records the elapsed time for processing templates into executions. This spec came sometime after that class was written.